cmd/restic/integration_test.go

```diff
@@ -1747,16 +1747,32 @@ func copyFile(dst string, src string) error {
 	if err != nil {
 		return err
 	}
-	defer srcFile.Close()
 
 	dstFile, err := os.Create(dst)
 	if err != nil {
+		// ignore subsequent errors
+		_ = srcFile.Close()
 		return err
 	}
-	defer dstFile.Close()
 
 	_, err = io.Copy(dstFile, srcFile)
-	return err
+	if err != nil {
+		// ignore subsequent errors
+		_ = srcFile.Close()
+		_ = dstFile.Close()
+	}
+
+	err = srcFile.Close()
+	if err != nil {
+		return err
```

Comment:
`dstFile` is not closed. The code might be more readable by using a name return value `err error` and then use a function as follows:
```go
defer func() {
   cerr := srcFile.Close()
   if err == nil {
       err = cerr
   }
}()

Reply:
you're right, I've added `_ = dstFile.Close()` in the `if`. I don't like using named returns as it is very easy to accidentally shadow the named return and thereby dropping errors :confused: